integration/airflow/openlineage/airflow/extractors/dbt_cloud_extractor.py

```diff
@@ -0,0 +1,248 @@
+# Copyright 2018-2022 contributors to the OpenLineage project
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import re
+import traceback
+
+# for concurrent processing
+from concurrent.futures import ThreadPoolExecutor
+from typing import Dict, List, Optional
+
+from openlineage.airflow.extractors.base import BaseExtractor, TaskMetadata
+from openlineage.client.facet import ErrorMessageRunFacet
+from openlineage.common.provider.dbt import DbtCloudArtifactProcessor, ParentRunMetadata
+
+from airflow.hooks.base import BaseHook
+from airflow.providers.dbt.cloud.hooks.dbt import DbtCloudHook
+
+log = logging.getLogger(__name__)
+
+
+class DbtCloudExtractor(BaseExtractor):
+    default_schema = "public"
+
+    def __init__(self, operator):
+        super().__init__(operator)
+
+    @classmethod
+    def get_operator_classnames(cls) -> List[str]:
+        return ["DbtCloudRunJobOperator", "DbtCloudJobRunSensor"]
```

Comment:
Can we support `DbtCloudRunJobOperatorAsync` too?

Reply:
> @JDarDagran can we remove unneeded data, like `macros` from `manifest.json`?

Done.

> Can we support `DbtCloudRunJobOperatorAsync` too?

There's no such thing in Airflow provider. The only thing is `deferrable` that paired with `wait_for_termination` set to True makes operator wait asynchronously.

Reply:
@JDarDagran https://github.com/astronomer/astronomer-providers/blob/main/astronomer/providers/dbt/cloud/operators/dbt.py#L12

Reply:
Okay, added both async operator and sensor from astronomer provider.